configure

```diff
@@ -20884,6 +20887,23 @@ $as_echo "#define SDL_VIDEO_DRIVER_RPI 1" >>confdefs.h
     fi
 }
 
+CheckRockchip()
+{
+    # Check whether --enable-video-rockchip was given.
+if test "${enable_video_rockchip+set}" = set; then :
+  enableval=$enable_video_rockchip;
```

Comment:
I see that enableval is set in other check functions just like this, but it's not really being used here (or in a lot of other places for that matter). I suppose to keep with consistency it's probably best to leave it, but I don't think it's needed. If you do leave it, then to mirror the logic in other checks you'll want to set enable_video_rockchip=yes in the else condition.

Reply:
CheckRockchip() is re-introduced in configure file here:
https://github.com/avafinger/SDL/blob/d6a25587b65aa85dd69301878e6dc92d29c02684/configure#L25816

and if you pass : --enable-video-kmsdrm --enable-video-rockchip --disable-video-x11
SDL_VIDEO_DRIVER_ROCKCHIP and SDL_HINT_VIDEO_X11_FORCE_EGL is set here:
https://github.com/avafinger/SDL/blob/d6a25587b65aa85dd69301878e6dc92d29c02684/configure#L20899

and you have configured for GBM and CLI.

I just tested it cloning the repo and doing:./configure --enable-video-kmsdrm --enable-video-rockchip --enable-video-x11 --disable-video-vulkan  --disable-video-wayland 

without invoking rm ./include/SDL_config.h and ./autogen.sh

that worked with RK3568 and RK3399 unless you tell me you can pass arguments to configure in the buildroot.

Reply:
This line:
https://github.com/avafinger/SDL/blob/d6a25587b65aa85dd69301878e6dc92d29c02684/configure#L20896
Should resemble this line:
https://github.com/avafinger/SDL/blob/d6a25587b65aa85dd69301878e6dc92d29c02684/configure#L20913

As when "${enable_video_rockchip+set}" = set, then enable_video_rockchip is blank, so conversely it needs to be set to yes.

Reply:
> that worked with RK3568 and RK3399 unless you tell me you can pass arguments to configure in the buildroot.

Sorry, I wanted to mean: you can **NOT** pass arguments.

If I change that line to = yes , my understanding is that if you run config without parameters you end up with:
| #define SDL_VIDEO_DRIVER_ROCKCHIP 1
| #define SDL_HINT_VIDEO_X11_FORCE_EGL 1

And this will break GLX for other platforms, like x64... because I have to disable GLX in X11 when SDL_VIDEO_DRIVER_ROCKCHIP is set. Unless SDL_HINT_VIDEO_X11_FORCE_EGL = 1 should be fixed.

the default is NO:
--enable-video-rockchip use Rockchip video driver with mali blob
                          [default=no]

Please, can you make it clear that you mean this:

```
CheckRockchip()
{
    # Check whether --enable-video-rockchip was given.
if test "${enable_video_rockchip+set}" = set; then :
  enableval=$enable_video_rockchip;
else
  enable_video_rockchip=**yes**
fi

    if test x$enable_video = xyes -a x$enable_video_rockchip = xyes; then

$as_echo "#define SDL_VIDEO_DRIVER_ROCKCHIP 1" >>confdefs.h
$as_echo "#define SDL_HINT_VIDEO_X11_FORCE_EGL 1" >>confdefs.h

    fi
}
```

Reply:
tested with GBM and CLI, broken with the latest commit. X11 works. i will put some debugging info and see where it breaks.
Looks like mali blob has different implementations for X11 and GBM.